modules/local/kraken2/main.nf

```diff
@@ -0,0 +1,39 @@
+process KRAKEN2 {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(reads), path(kraken2_db)
+
+    output:
+    tuple val(meta), path("*_kraken2_output.tsv"), path("*_kraken2_report.txt"), emit:report
+    path "versions.yml",                                                         emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def paired = meta.paired_end ? "--paired" : ""
```

Comment:
Nf-core uses `meta.single_end` to indiciate if something is single end. Could you use that variable in this expression instead of `paired_end`?

Reply:
Updated in: [7828098](https://github.com/phac-nml/speciesabundance/pull/5/commits/7828098d561a3c6f507876a2de471519adaddd75)
Thanks again for your help on this one!

Reply:
You're welcome. Thank you so much for all your work 😄